specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/DataConnectors.json

```diff
@@ -0,0 +1,1696 @@
+      },
+      "type": "object",
+      "x-ms-discriminator-value": "JwtToken",
+      "required": [
+        "tokenEndpoint",
+        "userName",
+        "password"
+      ]
+    },
+    "OAuthModel": {
+      "description": "Model for API authentication with OAuth2.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "properties": {
+        "authorizationCode": {
+          "type": "string",
+          "description": "The user's authorization code."
+        },
+        "clientSecret": {
+          "type": "string",
+          "description": "The Application (client) secret that the OAuth provider assigned to your app."
+        },
+        "clientId": {
+          "type": "string",
+          "description": "The Application (client) ID that the OAuth provider assigned to your app."
+        },
+        "isCredentialsInHeaders": {
+          "type": "boolean",
+          "x-nullable": true,
+          "default": false,
+          "description": "Indicating whether we want to send the clientId and clientSecret to token endpoint in the headers."
+        },
+        "scope": {
+          "type": "string",
+          "description": "The Application (client) Scope that the OAuth provider assigned to your app."
+        },
+        "redirectUri": {
+          "type": "string",
+          "description": "The Application redirect url that the user config in the OAuth provider."
+        },
+        "grantType": {
+          "type": "string",
+          "description": "The grant type, usually will be 'authorization code'."
+        },
+        "tokenEndpoint": {
+          "type": "string",
+          "description": "The token endpoint. Defines the OAuth2 refresh token."
+        },
+        "tokenEndpointHeaders": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The token endpoint headers."
+        },
+        "tokenEndpointQueryParameters": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The token endpoint query parameters."
+        },
+        "authorizationEndpoint": {
+          "type": "string",
+          "description": "The authorization endpoint."
+        },
+        "authorizationEndpointHeaders": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The authorization endpoint headers."
+        },
+        "authorizationEndpointQueryParameters": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The authorization endpoint query parameters."
+        },
+        "isJwtBearerFlow": {
+          "type": "boolean",
+          "description": "A value indicating whether it's a JWT flow."
+        },
+        "accessTokenPrepend": {
```

Comment:
>accessTokenPrepend [](http://example.com/codeflow?start=9&length=18)

can this be any string? can you convert it to an enum?

Reply:
PM decision was to have this value as string. Discussed with ARM in Preview review